app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/TravelCertificateHistory.kt

```diff
@@ -0,0 +1,422 @@
+import com.hedvig.android.core.icons.hedvig.normal.Info
+import com.hedvig.android.core.ui.clearFocusOnTap
+import com.hedvig.android.core.ui.dialog.ErrorDialog
+import com.hedvig.android.core.ui.infocard.VectorInfoCard
+import com.hedvig.android.core.ui.scaffold.HedvigScaffold
+import com.hedvig.android.data.travelcertificate.TravelCertificate
+import com.hedvig.android.feature.travelcertificate.CertificateHistoryEvent
+import com.hedvig.android.feature.travelcertificate.CertificateHistoryUiState
+import com.hedvig.android.feature.travelcertificate.CertificateHistoryViewModel
+import com.hedvig.android.feature.travelcertificate.data.TravelCertificateUri
+import hedvig.resources.R
+import kotlinx.datetime.LocalDate
+
+@Composable
+internal fun TravelCertificateHistoryDestination(
+  viewModel: CertificateHistoryViewModel,
+  onContinue: () -> Unit,
+  navigateUp: () -> Unit,
+  onShareTravelCertificate: (TravelCertificateUri) -> Unit,
+) {
+  val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+  TravelCertificateHistoryScreen(
+    reload = { viewModel.emit(CertificateHistoryEvent.RetryLoadReferralData) },
+    onCertificateClick = { url ->
+      viewModel.emit(CertificateHistoryEvent.DownloadCertificate(url))
+    },
+    onContinue = onContinue,
+    navigateUp = navigateUp,
+    onShareTravelCertificate = onShareTravelCertificate,
+    uiState = uiState,
+  )
+}
+
+@Composable
+private fun TravelCertificateHistoryScreen(
+  reload: () -> Unit,
+  onCertificateClick: (String) -> Unit,
+  onContinue: () -> Unit,
+  navigateUp: () -> Unit,
+  onShareTravelCertificate: (TravelCertificateUri) -> Unit,
+  uiState: CertificateHistoryUiState,
+) {
+  var showBottomSheet by remember { mutableStateOf(false) }
+  val explanationSheetState = rememberModalBottomSheetState(true)
+  if (showBottomSheet) {
+    TravelCertificateInfoBottomSheet(
+      onDismiss = { showBottomSheet = false },
+      sheetState = explanationSheetState,
+    )
+  }
+
+  when (uiState) {
+    CertificateHistoryUiState.FailureDownloadingCertificate -> {
+      ErrorDialog(
+        title = stringResource(id = R.string.general_error),
+        message = stringResource(id = R.string.travel_certificate_downloading_error),
+        onDismiss = {
+          reload()
+        },
+      )
+    }
+
+    CertificateHistoryUiState.FailureDownloadingHistory -> {
+      HedvigScaffold(
+        navigateUp = navigateUp,
+        modifier = Modifier.clearFocusOnTap(),
+        topAppBarText = stringResource(id = R.string.PROFILE_ROW_TRAVEL_CERTIFICATE),
+        topAppBarActions = {
+          IconButton(
+            onClick = { showBottomSheet = true },
+            modifier = Modifier.size(40.dp),
+          ) {
+            Icon(
+              imageVector = Icons.Hedvig.Info,
+              contentDescription = stringResource(R.string.REFERRALS_INFO_BUTTON_CONTENT_DESCRIPTION),
+              modifier = Modifier.size(24.dp),
+            )
+          }
+        },
+      ) {
```

Comment:
I wonder if in the failure scenario we can just show an error in the middle of the screen with the ability to retry loading and not show the top app bar with the info button and so on? I know the design does not specify it itself, so I wonder how this screen looks like with the info at the top but with error state in the middle.


If we do in fact want to keep the scaffold here anyway, I wonder then if the scaffold can be specified above here, so that it always show, and only the content of the scaffold changes depending on if we got a success or a failed state and so on. That way no need to call the scaffold twice in each case.

But if you've tried smth like that and think it doens't work here for some reason I'd also be happy to discuss that.